amr-wind/turbulence/RANS/KOmegaSST.cpp

```diff
@@ -182,24 +183,53 @@ void KOmegaSST<Transport>::update_turbulent_viscosity(const FieldState fstate)
 
                     f1_arr(i, j, k) = tmp_f1;
 
+                    // For TKE equation:
+                    shear_prod_arr(i, j, k) = amrex::min<amrex::Real>(
+                        amrex::max<amrex::Real>(
+                            mu_arr(i, j, k) * tmp4 * tmp4, 0.0),
+                        10.0 * beta_star * rho_arr(i, j, k) * tke_arr(i, j, k) *
+                            sdr_arr(i, j, k));
+
                     diss_arr(i, j, k) = -beta_star * rho_arr(i, j, k) *
                                         tke_arr(i, j, k) * sdr_arr(i, j, k);
+
                     tke_lhs_arr(i, j, k) = 0.5 * beta_star * rho_arr(i, j, k) *
                                            sdr_arr(i, j, k) * deltaT;
 
-                    shear_prod_arr(i, j, k) = amrex::min<amrex::Real>(
-                        mu_arr(i, j, k) * tmp4 * tmp4,
-                        10.0 * beta_star * rho_arr(i, j, k) * tke_arr(i, j, k) *
-                            sdr_arr(i, j, k));
+                    // For SDR equation:
+                    amrex::Real production_omega =
+                        rho_arr(i, j, k) * alpha *
+                        amrex::min<amrex::Real>(
+                            tmp4 * tmp4, 10.0 * beta_star * sdr_arr(i, j, k) *
+                                             sdr_arr(i, j, k));
 
-                    sdr_lhs_arr(i, j, k) = 0.5 * rho_arr(i, j, k) * beta *
-                                           sdr_arr(i, j, k) * deltaT;
-                    sdr_src_arr(i, j, k) =
-                        rho_arr(i, j, k) * alpha * tmp4 * tmp4 +
+                    amrex::Real cross_diffusion =
                         (1.0 - tmp_f1) * 2.0 * rho_arr(i, j, k) * sigma_omega2 *
-                            gko / (sdr_arr(i, j, k) + 1e-15);
-                    sdr_diss_arr(i, j, k) = -rho_arr(i, j, k) * beta *
-                                            sdr_arr(i, j, k) * sdr_arr(i, j, k);
+                        gko / (sdr_arr(i, j, k) + 1e-15);
+
+                    if (diff_type == DiffusionType::Crank_Nicolson) {
+
+                        sdr_src_arr(i, j, k) = production_omega;
```

Comment:
You can use ternary expressions and make it easier to parse. For example: 
` sdr_src_arr(i, j, k) = production_omega + (diff_type != DiffusionType::Crank_Nicolson)? cross_diffusion : 0.0:` 
Can you do that for the others too?

Reply:
The change to ternary expression can make code readability a bit confusing.